vault/src/system.rs

```diff
@@ -581,8 +581,9 @@ impl VaultService {
             if let Some(collateral) = self.config.auto_register_with_collateral {
                 tracing::info!("[{}] Automatically registering...", vault_id.pretty_printed());
                 let public_key = bitcoin_core_with_wallet.get_new_public_key().await?;
+                let free_balance = self.btc_parachain.get_free_balance(collateral_currency).await?;
                 self.btc_parachain
-                    .register_vault(&vault_id, collateral, public_key)
+                    .register_vault(&vault_id, std::cmp::min(collateral, free_balance), public_key)
```

Comment:
What if the collateral currency is KINT? We want to make sure there is enough to pay for transaction fees after registering. Although, I guess the register_vault would fail in that case anyway. Should we catch that case and prevent the transaction fee from being wasted?

Reply:
I don't think we should focus on that right now, firstly because it'll be a while before KINT is accepted as a collateral currency and secondly because we will also implement the "bring your own gas" model so fees will be payable in every currency anyway - which requires more thinking outside of this PR.